lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,7 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A(([a-z0-9])|((\.?[\w\-\@])([\w\-\@]|[\ ][\w\-\@]|[\/][\w\-\@\.]|[\.][\w\-\@]+)*))\z/i
```

Comment:
I wasn't suggesting posix compliance was the end goal; just that we should probably aim for simplicity if possible. Maybe breaking it down makes sense? something like

1. return invalid on relative paths `/..` and `/.`
2. validate file is legit by creating one in the temp directory

or 
1. return invalid on relative paths
2. a simpler regex that validates there are `[filename_allowed_chars]+[.]?[filename_allowed_chars]*` guess this wouldn't allow for `foo.bar.biz`...